src/_blog/metrics-in-go-libp2p.md

```diff
@@ -0,0 +1,137 @@
+---
+tags:
+- metrics
+- prometheus
+title: Metrics in go-libp2p
+description:
+date: 2023-06-15
+permalink: "/2023-06-15-metrics-in-go-libp2p/"
+author: Sukun Tarachandani
+---
+
+# Metrics in go-libp2p
+
+## Introduction
+
+Libp2p is the core networking component for projects and so it is important to be able to observe the state of its components. To that effect, we've been adding metrics to the various components over the last few months. They've already helped us debug some issues and helped with the development of smart dialing. 
+
+## Why Prometheus?
+
+After deliberations between prometheus, opencensus, and opentelemetry, we decided to use prometheus. The details of the discussion can be found [here](https://github.com/libp2p/go-libp2p/issues/1356). To summarise, prometheus was performant and ubiquitious. This allowed us to add metrics without worrying too much about performance. We also ensured that tracking metrics didn't put too much pressure on the garbage collector, something that we'd found to be an [issue with opencensus](https://github.com/libp2p/go-libp2p/issues/1955).
```

Comment:
```suggestion
We were first faced with the question of choosing a metrics collection and monitoring system. Among our choices were prometheus, opencensus, and opentelemetry.

After deliberations,  we decided to use prometheus. The details of the discussion can be found [here](https://github.com/libp2p/go-libp2p/issues/1356). To summarise, prometheus was performant and ubiquitious. This allowed us to add metrics without worrying too much about performance. We also ensured that tracking metrics didn't put too much pressure on the garbage collector, something that we'd found to be an [issue with opencensus](https://github.com/libp2p/go-libp2p/issues/1955).
```

I would rephrase the intro like so.

Is it also worth describing why we didn't choose others? For example you mentioned the issue with OpenCensus. For OpenTelemetry should you mention that we didn't want to depend on alpha software?

Also is there anything here to say about considerations for users i.e. would choosing Prometheus make it easier to users of go-libp2p to plug in Grafana?

Reply:
I added a line about Grafana.